opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StatsFragmentInteractor.java

```diff
@@ -0,0 +1,133 @@
+package org.smartregister.reveal.interactor;
+
+import net.sqlcipher.database.SQLiteDatabase;
+import net.sqlcipher.Cursor;
+import net.sqlcipher.SQLException;
+
+import org.smartregister.repository.BaseRepository;
+import org.smartregister.reveal.application.RevealApplication;
+import org.smartregister.reveal.contract.StatsFragmentContract;
+import org.smartregister.reveal.util.AppExecutors;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.SYNC_STATUS;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.VALIDATION_STATUS;
+import static org.smartregister.reveal.util.Constants.SyncInfo.INVALID_CLIENTS;
+import static org.smartregister.reveal.util.Constants.SyncInfo.INVALID_EVENTS;
+import static org.smartregister.reveal.util.Constants.SyncInfo.NULL_EVENT_SYNC_STATUS;
+import static org.smartregister.reveal.util.Constants.SyncInfo.SYNCED_CLIENTS;
+import static org.smartregister.reveal.util.Constants.SyncInfo.SYNCED_EVENTS;
+import static org.smartregister.reveal.util.Constants.SyncInfo.TASK_UNPROCESSED_EVENTS;
+import static org.smartregister.reveal.util.Constants.SyncInfo.UNSYNCED_CLIENTS;
+import static org.smartregister.reveal.util.Constants.SyncInfo.UNSYNCED_EVENTS;
+import static org.smartregister.reveal.util.Constants.SyncInfo.VALID_CLIENTS;
+import static org.smartregister.reveal.util.Constants.SyncInfo.VALID_EVENTS;
+
+public class StatsFragmentInteractor implements StatsFragmentContract.Interactor {
+
+    private AppExecutors appExecutors;
+
+    private SQLiteDatabase database;
+
+    private StatsFragmentContract.Presenter presenter;
+
+
+    public StatsFragmentInteractor(StatsFragmentContract.Presenter presenter) {
+        this.presenter = presenter;
+        appExecutors = RevealApplication.getInstance().getAppExecutors();
+        database = RevealApplication.getInstance().getRepository().getReadableDatabase();
+    }
+
+    @Override
+    public void fetchECSyncInfo() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'fetchECSyncInfo()' has an NPath complexity of 385](https://app.codacy.com/app/OpenSRP/opensrp-client-reveal/pullRequest?prid=4587928)